src/writers/json-writer.cc

```diff
@@ -0,0 +1,101 @@
+namespace std { class type_info; }
+
+#include <reporter.hh>
+#include <file-parser.hh>
+#include <configuration.hh>
+#include <writer.hh>
+#include <utils.hh>
+
+#include <string>
+#include <list>
+#include <unordered_map>
+#include <iostream>
+#include <fstream>
+
+#include "writer-base.hh"
+
+using namespace kcov;
+
+class JsonWriter : public WriterBase
+{
+public:
+	JsonWriter(IFileParser &parser, IReporter &reporter,
+			const std::string &outFile) :
+		WriterBase(parser, reporter),
+		m_outFile(outFile)
+	{
+	}
+
+	void onStartup()
+	{
+	}
+
+	void onStop()
+	{
+	}
+
+	void write()
+	{
+		std::ofstream out(m_outFile);
+
+		// Output directory not writable?
+		if (!out.is_open())
+			return;
+
+		IConfiguration &conf = IConfiguration::getInstance();
+		unsigned int nExecutedLines = 0;
+		unsigned int nCodeLines = 0;
+
+		setupCommonPaths();
+
+		for (FileMap_t::const_iterator it = m_files.begin();
+				it != m_files.end();
+				++it) {
+			File *file = it->second;
+			nCodeLines += file->m_codeLines;
+			nExecutedLines += file->m_executedLines;
```

Comment:
Now that I think of it, I think it might be that the File object is used by all writers, so technically, only one need to update the stats. Of course, that shouldn't be relied on, so it should really be refactored to be handled outside. (Sorry about that!)

Anyway, I think the best way is to get the json writer working, and then refactor them all afterwards. There are probably other things which should be changed as well, now that there will be quite a few different writers.